data/provider/src/main/java/tech/pegasys/artemis/api/ChainDataProvider.java

```diff
@@ -139,6 +146,36 @@ CombinedChainDataClient getCombinedChainDataClient() {
         .exceptionally(err -> Optional.empty());
   }
 
+  public Optional<Attestation> getUnsignedAttestationAtSlot(
+      UnsignedLong slot, Integer committeeIndex) {
+    if (!isStoreAvailable()) {
+      return Optional.empty();
+    }
+    if (isFinalized(slot)) {
+      throw new IllegalArgumentException(
+          String.format("Slot %s is finalized, no attestation will be created.", slot.toString()));
+    }
+    Optional<BeaconBlock> block = chainStorageClient.getBlockBySlot(slot);
+    if (block.isEmpty()) {
+      return Optional.empty();
+    }
+
+    tech.pegasys.artemis.datastructures.state.BeaconState state =
+        chainStorageClient.getBestBlockRootState();
+    int committeeCount = get_committee_count_at_slot(state, slot).intValue();
+    if (committeeIndex < 0 || committeeIndex > committeeCount) {
+      throw new IllegalArgumentException(
+          "Invalid committee index provided - expected between 0 and " + (committeeCount - 1));
```

Comment:
what happens if it's == committeeCount? This msg doesn't quite match the if statement

Reply:
i'll fix this, but i'll also look at an upper boundary test, it looks like its not done if that check is wrong like that.